### PR TITLE
Clean gengen fixture data before generation

### LINKED ISSUE #2861: Remove contents fo `fixtures/[test/live]` during build prior to gengen
### Description

We load some data from disk in an `init` method in the fixtures package. This can result in a panic if the data-structure used to generate the data changes as we can no longer read it back.

This was discovered in #2859.

We should remove this data during build prior to running gengen so avoid this conflict in the future.

### Acceptance criteria

- Contents of `fixtures/[test/live]` are remove prior to a call to running gengen

### Where to begin

https://github.com/filecoin-project/go-filecoin/blob/master/build/main.go#L172-L189

### PR DESCRIPTION
**Pull request by travisperson**
Closes #2861

### CODE REVIEW

File: build/main.go [hunk 1]
```diff
@@ -169,15 +171,62 @@ func forceBuildFC() {
 	}...))
 }
 
+func cleanDirectory(dir string, ignoredots bool) error {
```
Comment (frrist) [inline build/main.go]: Can you add a small comment stating that we use this method (and not https://golang.org/pkg/os/#RemoveAll) to preserve the `.gitkeep` file? (or could rename the method to reflect that).

Is the `.gitkeep` necessary?

Comment (travisperson) [reply build/main.go]: Ya, the `.gitkeep` is necessary if we want the directory to exist on code checkout.

File: build/main.go [hunk 2]
```diff
@@ -169,15 +171,66 @@ func forceBuildFC() {
 	}...))
 }
 
+// cealnDirectory removes the child of a directly wihtout removing the directory itself, unlike `RemoveAll`.
```
Comment (anorth) [inline build/main.go]: `cleanDirectory`

File: build/main.go [hunk 3]
```diff
@@ -169,15 +171,66 @@ func forceBuildFC() {
...
+	for _, file := range files {
+		fname := file.Name()
+		if ignoredots && []rune(fname)[0] == '.' {
+			continue
+		}
+
+		fpath := filepath.Join(dir, fname)
+
+		fmt.Println("Removing", fpath)
+		if err := os.Remove(fpath); err != nil {
```
Comment (anorth) [inline build/main.go]: What if fpath is also a directory? This should probably RemoveAll.

File: build/main.go [hunk 4]
```diff
@@ -169,15 +171,66 @@ func forceBuildFC() {
...
+		}
+	}
+
+	return nil
+}
+
 func generateGenesis() {
 	log.Println("Generating genesis...")
+
+	liveFixtures, err := filepath.Abs("./fixtures/live")
```
Comment (anorth) [inline build/main.go]: Factor out the duplicated path names (line 219)?

**Pull request merged**